maven-core/src/main/java/org/apache/maven/execution/DefaultMavenExecutionRequest.java

```diff
@@ -1051,16 +1057,44 @@ public MavenExecutionRequest setToolchains(Map<String, List<ToolchainModel>> too
         return this;
     }
 
+    @Deprecated
     @Override
     public void setMultiModuleProjectDirectory(File directory) {
         this.multiModuleProjectDirectory = directory;
     }
 
+    @Deprecated
     @Override
     public File getMultiModuleProjectDirectory() {
         return multiModuleProjectDirectory;
     }
 
+    @Override
+    public Path getRootdir() {
+        if (rootdir == null) {
+            throw new IllegalArgumentException("Unable to find the root directory. "
```

Comment:
This needs to be documented since it looks like a hard requirement now.

Reply:
The goal is to throw an exception when actually used.  So it's not a hard requirement unless you actually use it somehow in a plugin or in a model, in which case, yes, it becomes a hard requirement.

Reply:
But rather fail early than after 30 min of a build?

Reply:
But we can't know early if someone uses the `${session.rootdir}`or `${rootdir}` in the model, profile activation or a plugin configuration.  If we fail early, this will mean that all projects will be required to add this `.mvn` direction.  Note that we already have a warning if this directory does not exist.

Reply:
Accepted